### ISSUE TITLE
Inpaint with Fooocus, request sdxl model every time at KSampler. Cost extra time

### ISSUE THREAD

**Issue by jarheadjoe** [Acly/comfyui-inpaint-nodes#64]

**Comment by Acly** [comment]
It doesn't, unless you change the inputs like image/mask.

**Comment by jarheadjoe** [comment]
> It doesn't, unless you change the inputs like image/mask.

Yes, when change input image, it requests to load new model. But in normal i2i or inpaint workflow, if will not request new model when change inputs. I think the problem is in applyFoocusInpaint node: when change input image, the node calculate the feature and add to the model directly and return the model, which makes comfyui think it should load a new model. I think the node ought to to split to AssembleFoocusModel Node where it adds patch block to base model and ApplyFoocusInpaint where it feeds image and mask to the new model.

**Comment by Acly** [comment]
Feeding the image/mask to the new model also changes the model. Attaching the image/mask to the model is the source of the problem, it would make more sense if they were passed in while sampling. I don't see a way to implement it that way without changes in base ComfyUI though, as it doesn't pass the latent to the patch function.

**Comment by jarheadjoe** [comment]
Maybe write a custom load node to load patch() and lora() as model and add it to base model to become a inpaint model
